### ISSUE TITLE
Duplicate slot warning

### ISSUE THREAD

**Issue by blocka** [vuejs/vue#8546]
### Version
2.5.16

### Reproduction link
[https://codesandbox.io/s/vn7jz1l1kl](https://codesandbox.io/s/vn7jz1l1kl)

### Steps to reproduce
Click the button that says "refresh"

### What is expected?
Nothing should happen

### What is actually happening?
[Vue warn]: Duplicate presence of slot "default" found in the same render tree - this will likely cause render errors.

---
I have a component `B` which accepts a scopedSlot, and a component `A` which accepts a slot. `A` is passing in it's slot into the scopedSlot of `B`

```
<template slot-scope="scope">
          <slot></slot>
</template>
```

When the component `B` gets rerendered, I see this warning in the console. `B` passes a method to the scoped slot which will cause it to rerender when called.

**Comment by Justineo** [comment]
As I understand this won’t be a problem if #8496 is resolved.

**Comment by blocka** [comment]
I'm not sure there is a connection. I'm not rendering a slot here multiple times. Vue, for some reason, thinks that it's already been rendered.

**Comment by Justineo** [comment]
Haven’t dug into the code yet but I think maybe it’s because scoped slots are rendered like a function call so you keep rendering the slot inside it each time you  rerender the component.

**Comment by webther** [comment]
We can use the 'Named Slots' to avoid the duplicate slots.

`<A><template slot-scope="slotProps" slot="hello">blarg</template></A>`

`<slot name="hello"></slot>`

Here is a working example: https://codesandbox.io/s/66ny97z5r

**Comment by blocka** [comment]
You're comment confused the heck out of me until I realized that I had something in there that I didn't mean to have.

The API for `A` should only take a default slot, and that's it (certainly not a scoped slot). Anything else is just an awkward looking api. I've updated the sandbox.

```html
<A>Blarg</A>
```

Is the API I am after. That default slots gets passed into the scoped slot exposed by `B`.

**Comment by yyx990803** [comment]
This should have been fixed by 530ca1b2 which removes the no-duplicated-slot restriction.
